organisaatio-ui/src/components/Taulukot/NormaaliTaulukko.tsx

```diff
@@ -0,0 +1,296 @@
+import * as React from 'react';
+import styles from './NormaaliTaulukko.module.css';
+import {
+    useGlobalFilter,
+    useExpanded,
+    usePagination,
+    useTable,
+    useSortBy,
+    useFilters,
+    Column,
+    TableInstance,
+} from 'react-table';
+import Button from '@opetushallitus/virkailija-ui-components/Button';
+import Input from '@opetushallitus/virkailija-ui-components/Input';
+import searchIcon from '@iconify/icons-fa-solid/search';
+import Select from '@opetushallitus/virkailija-ui-components/Select';
+import { Icon } from '@iconify/react';
+import chevronLeft from '@iconify/icons-fa-solid/chevron-left';
+import chevronRight from '@iconify/icons-fa-solid/chevron-right';
+import { useContext } from 'react';
+import { KoodistoContext, LanguageContext } from '../../contexts/contexts';
+import { useState } from 'react';
+import { Ryhma, SelectOptionType, YhteystietoTyyppi } from '../../types/types';
+import { ValueType } from 'react-select';
+
+// TODO yhtenäistä!
+const mapPaginationSelectors = (index) => {
+    if (index < 3) return [0, 5];
+    return [index - 2, index + 3];
+};
+
+export type Props = {
+    ryhmatData?: Ryhma[] | undefined;
+    ryhmatColumns?: Column<Ryhma>[] | undefined;
+    yhteystietoTyypitData?: YhteystietoTyyppi[] | undefined;
+    yhteystietotyypitColumns?: Column<YhteystietoTyyppi>[] | undefined;
+    useHakuFiltteri?: boolean;
+};
+
+export type FiltteritProps = {
+    setFilter: (columnId: string, updater: any) => void;
+    setGlobalFilter: (filterValue: any) => void;
+    globalFilter: string;
+};
+
+export const Hakufiltterit = ({ setFilter, globalFilter, setGlobalFilter }: FiltteritProps) => {
+    const { i18n } = useContext(LanguageContext);
+    const { ryhmaTyypitKoodisto, kayttoRyhmatKoodisto, ryhmanTilaKoodisto } = useContext(KoodistoContext);
+    const [tyyppiFiltteri, setTyyppiFiltteri] = useState<
+        ValueType<SelectOptionType>[] | ValueType<SelectOptionType> | undefined
+    >();
+    const [kayttoRyhmatFiltteri, setKayttoRyhmatFiltteri] = useState<ValueType<SelectOptionType>[] | undefined>();
+    const [tilaFiltteri, setTilaFiltteri] = useState<ValueType<SelectOptionType>[] | undefined>();
+
+    const ryhmatyypitOptions = ryhmaTyypitKoodisto.koodit().map((k) => {
+        return { value: k.arvo, label: k.nimi['fi'] || k.nimi['sv'] || k.nimi['en'] || '' };
```

Comment:
...and here we see that localization kludge again for label? seems to be repeating pattern. what if user prefers some other locale as default?